gprofiler/metadata/application_identifiers.py

```diff
@@ -289,25 +290,27 @@ def get_app_id(self, process: Process) -> Optional[str]:
 
 # Please note that the order matter, because the FIRST matching identifier will be used.
 # so when adding new identifiers pay attention to the order.
-_PYTHON_APP_IDENTIFIERS = [
-    _GunicornTitleApplicationIdentifier(),
-    _GunicornApplicationIdentifier(),
-    _UwsgiApplicationIdentifier(),
-    _CeleryApplicationIdentifier(),
-    _PySparkApplicationIdentifier(),
-    _PythonModuleApplicationIdentifier(),
-]
-
-_JAVA_APP_IDENTIFIERS: List[_ApplicationIdentifier] = [
-    _JavaJarApplicationIdentifier(),
-]
+_IDENTIFIERS_MAP: Dict[str, List[_ApplicationIdentifier]] = {
+    "python": [
+        _GunicornTitleApplicationIdentifier(),
+        _GunicornApplicationIdentifier(),
+        _UwsgiApplicationIdentifier(),
+        _CeleryApplicationIdentifier(),
+        _PySparkApplicationIdentifier(),
+        _PythonModuleApplicationIdentifier(),
+    ],
+    "java": [
+        _JavaJarApplicationIdentifier(),
+    ],
+}
 
 
 def set_enrichment_options(enrichment_options: EnrichmentOptions) -> None:
     _ApplicationIdentifier.enrichment_options = enrichment_options
 
 
-def get_app_id(process: Process, identifiers: List[_ApplicationIdentifier]) -> Optional[str]:
+@functools.lru_cache(4096)  # TODO: arbitrary cache size
+def get_app_id(process: Process, runtime: Union[Literal["python"], Literal["java"]]) -> Optional[str]:
```

Comment:
This is gonna mess with https://github.com/Granulate/gprofiler/pull/393, but k, we'll solve that